lnwallet/wallet_test.go

```diff
@@ -0,0 +1,176 @@
+// expected amount.
+// NOTE: coinSelect will always attempt to add a change output, so we must
+// account for this in the tests.
+func TestCoinSelect(t *testing.T) {
+	t.Parallel()
+
+	const feeRate = SatPerKWeight(100)
+	const dust = btcutil.Amount(100)
+
+	type testCase struct {
+		name        string
+		outputValue btcutil.Amount
+		coins       []*Utxo
+
+		expectedInput  []btcutil.Amount
+		expectedChange btcutil.Amount
+		expectErr      bool
+	}
+
+	testCases := []testCase{
+		{
+			// We have 1.0 BTC available, and wants to send 0.5.
+			// This will obviously lead to a change output of
+			// almost 0.5 BTC.
+			name: "big change",
+			coins: []*Utxo{
+				{
+					AddressType: WitnessPubKey,
+					Value:       1 * btcutil.SatoshiPerBitcoin,
+				},
+			},
+			outputValue: 0.5 * btcutil.SatoshiPerBitcoin,
+
+			// The one and only input will be selected.
+			expectedInput: []btcutil.Amount{
+				1 * btcutil.SatoshiPerBitcoin,
+			},
+			// Change will be what's left minus the fee.
+			expectedChange: 0.5*btcutil.SatoshiPerBitcoin - fee(feeRate, 1, true),
+		},
+		{
+			// We have 1 BTC available, and we want to send 1 BTC.
+			// This should lead to an error, as we don't have
+			// enough funds to pay the fee.
+			name: "nothing left for fees",
+			coins: []*Utxo{
+				{
+					AddressType: WitnessPubKey,
+					Value:       1 * btcutil.SatoshiPerBitcoin,
+				},
+			},
+			outputValue: 1 * btcutil.SatoshiPerBitcoin,
+			expectErr:   true,
+		},
+		{
+			// We have a 1 BTC input, and want to create an output
+			// as big as possible, such that the remaining change
+			// will be dust.
+			name: "dust change",
+			coins: []*Utxo{
+				{
+					AddressType: WitnessPubKey,
+					Value:       1 * btcutil.SatoshiPerBitcoin,
+				},
+			},
+			// We tune the output value by subtracting the expected
+			// fee and a small dust amount.
+			outputValue: 1*btcutil.SatoshiPerBitcoin - fee(feeRate, 1, true) - dust,
+
+			expectedInput: []btcutil.Amount{
+				1 * btcutil.SatoshiPerBitcoin,
+			},
+
+			// Change will the dust.
+			expectedChange: dust,
+		},
+		{
+			// We have a 1 BTC input, and want to create an output
+			// as big as possible, such that there is nothing left
+			// for change.
+			name: "no change",
+			coins: []*Utxo{
+				{
+					AddressType: WitnessPubKey,
+					Value:       1 * btcutil.SatoshiPerBitcoin,
+				},
+			},
+			// We tune the output value to be the maximum amount
+			// possible, leaving just enough for fees.
+			outputValue: 1*btcutil.SatoshiPerBitcoin - fee(feeRate, 1, true),
+
+			expectedInput: []btcutil.Amount{
+				1 * btcutil.SatoshiPerBitcoin,
+			},
+			// We have just enough left to pay the fee, so there is
+			// nothing left for change.
+			// TODO(halseth): currently coinselect estimates fees
+			// assuming a change output.
+			expectedChange: 0,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
Can bind the loop variable `test := test`, then use `t.Paralle()` within the `Run` closure.

Reply:
fixed: https://github.com/lightningnetwork/lnd/pull/3165/commits/53f5e10b856323356115c9ac8df9792b869f4c4a